src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -17,6 +17,9 @@ namespace ray {
 typedef std::function<std::shared_ptr<WorkerLeaseInterface>(const rpc::Address &)>
     LeaseClientFactoryFn;
 
+// TaskQueues are keyed on resource shape and task dependencies.
+using TaskQueueKey = std::pair<SchedulingClass, std::vector<ObjectID>>;
```

Comment:
I might missed some contexts here: as I understand only the `ResourceSet` in `SchedulingClass` matters when we choose a worker lease, it seems like the `FunctionDescriptor` is also used because of the fair share scheduling? @ericl 

I don't quite understand why task dependencies are included here. Are we going to only use direct task calls for tasks with the same args? That seems to be quite limited usage.

Reply:
Note the vast majority of tasks will have no plasma args due to argument inlining. So this is only falling back to raylet scheduling for the rare tasks that see plasma IDs.

Reply:
okay, that makes sense. thanks! 

Currently when do we use by-reference args when submitting a direct task (direct actor or direct general task)?  Is by-reference arg only used to workaround the 4MB limitation of grpc?